src/Test/Perf/util/TraceManager_util.csx

```diff
@@ -0,0 +1,231 @@
+#load "ScenarioGenerator_util.csx"
+
+using System;
+using System.Diagnostics;
+using System.IO;
+
+interface ITraceManager
+{
+    int Iterations { get; }
+
+    void Cleanup();
+    void EndEvent();
+    void EndScenario();
+    void EndScenarios();
+    void ResetScenarioGenerator();
+    void Setup();
+    void Start();
+    void StartEvent();
+    void StartScenario(string scenarioName, string processName);
+    void Stop();
+    void WriteScenariosFileToDisk();
+}
+
+class TraceManagerFactory
+{
+    public static ITraceManager GetTraceManager(
+        int iterations = 3,
+        string cpcFolderPath = @"%SYSTEMDRIVE%\CPC",
+        string scenarioPath = @"%SYSTEMDRIVE%\CPC")
+    {
+        var cpcFullPath = Path.Combine(Environment.ExpandEnvironmentVariables(cpcFolderPath), "CPC.exe");
+        if (File.Exists(cpcFullPath))
+        {
+            var expandedScenarioPath = Environment.ExpandEnvironmentVariables(scenarioPath);
+            return new TraceManager(iterations, cpcFullPath, expandedScenarioPath);
+        }
+        else
+        {
+            return new NoOpTraceManager(iterations);
+        }
+    }
+}
+
+/// This is the Trace Manager that will be used when there is no CPC found in the machine
+/// All operations are NoOp
+class NoOpTraceManager : ITraceManager
+{
+    private readonly int _iterations;
+    public NoOpTraceManager(int iterations)
+    {
+        _iterations = iterations;
+    }
+
+    public int Iterations
+    {
+        get
+        {
+            return _iterations;
+        }
+    }
+
+    public void Cleanup()
+    {
+    }
+
+    public void EndEvent()
+    {
+    }
+
+    public void EndScenario()
+    {
+    }
+
+    public void EndScenarios()
+    {
+    }
+
+    public void ResetScenarioGenerator()
+    {
+    }
+
+    public void Setup()
+    {
+    }
+
+    public void Start()
+    {
+    }
+
+    public void StartEvent()
+    {
+    }
+
+    public void StartScenario(string scenarioName, string processName)
+    {
+    }
+
+    public void Stop()
+    {
+    }
+
+    public void WriteScenariosFileToDisk()
+    {
+    }
+}
+
+class TraceManager : ITraceManager
+{
+    private readonly ScenarioGenerator _scenarioGenerator;
+
+    private string _cpcPath = "CPC.exe";
```

Comment:
Like mentioned below, this value doesn't matter. This stayed because of a refactoring. I will modify it